src/Main.elm

```diff
@@ -24,48 +28,170 @@ sampleUrl =
 main : Program Never
 main =
     App.program
-        { init = init , view = view , update = update , subscriptions = subscriptions }
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
 
 
 -- Subscriptions
 
 subscriptions : Model -> Sub Msg
 subscriptions model =
-    Sub.none
+  Mouse.moves (\{x, y} -> Position x y)
 
 
 -- Model
 
-type alias Model = GeoJson
+type alias Model =
+  { map: GeoJson
+  , mousePosition : (Int, Int)
+  , prevPosition : (Int, Int)
+  , dragging : Bool
+  , topLeft : Coordinate
+  , topRight : Coordinate
+  , bottomLeft : Coordinate
+  , bottomRight : Coordinate
+  , zoomLevel : Int
+  }
+
+type alias Coordinate =
+  { lat : Int
+  , lng : Int
+  }
+
+
+type alias WheelEvent =
+  { deltaX : Float
+  , deltaY : Float
+  , deltaZ : Float
+  }
+
 
 init : ( Model, Cmd Msg )
 init =
-    ( ( (Geometry(Point(0,0,[]) ) ), Nothing ) , fetchPerform sampleUrl )
+    ( { map = ( (Geometry(Point(0,0,[]) ) ), Nothing )
```

Comment:
there's a few other places with this too, I don't want to add comments everywhere, if you notice 'em update 'em